### ISSUE TITLE
Optionally ignore parts of commands

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#299]
### Problem

```r
plan <- drake_plan(
  datasets = mclapply(1:8, get_dataset, mc.cores = 8),
  analyses = lapply(datasets, analyze_dataset),
  summaries = lapply(analyses, summarize_analysis)
)
```

What if we want to change `mc.cores`? The content of `datasets` won't change, but `drake` will think `datasets` is out of date.

### Solution: a new `ignore()` function

```r
plan <- drake_plan(
  datasets = mclapply(1:8, get_dataset, mc.cores = ignore(8)),
  analyses = lapply(datasets, analyze_dataset),
  summaries = lapply(analyses, summarize_analysis)
)
```

`Drake` should not react when 8 is changed to something else. We can make this happen by modifying [`standardize_command()`](https://github.com/ropensci/drake/blob/e87f05ad70a9c605df885c07df1bc5fa484db352/R/commands.R#L64).  Also, `drake` should not reproducibly track any symbols mentioned inside `ignore()`, so the code analysis needs to change.

**Comment by krlmlr** [comment]
I think this is orthogonal to within-target parallelism, but a very useful feature anyway. Related: An `always()` function that will make sure the code is always run.

Can we name these functions `drake_isolate()` and `drake_xxx()` (in analogy to Shiny, perhaps with `xxx` = `always`)?

**Comment by wlandau-lilly** [comment]
What would an `always()` function do that is different from the `"always"` trigger?

We could consider `drake_isolate()`, but I think `isolate()` is most similar to the `file_in()`, `file_out()`, and `knitr_in()` functions, none of which has a `drake_` prefix.

**Comment by krlmlr** [comment]
`always()` would work pretty much the same as an `"always"` trigger, but you would be able to express it in a `drake_plan()`.

`isolate()` will collide with *shiny*, not sure if we need to care about that.

**Comment by wlandau** [comment]
Good point on `isolate()`. I still like the word `ignore()`, and I am leaning away from `always()` because it puts us in a situation where changing the trigger invalidates the target. We would also have to handle the case where `always()` appears in a command and the explicit entry in the `trigger` column is something else.

**Comment by krlmlr** [comment]
Can we prune the `always()` call during code analysis? I'd think `always()` either in the command or in the `trigger` column wins.

How do we create the `trigger` column in `drake_plan()`

**Comment by wlandau** [comment]
> Can we prune the always() call during code analysis?

Pruning `always()` would cover it.

> I'd think always() either in the command or in the trigger column wins.

Not sure I understand this sentence. Do you mean to say that the `trigger` column takes precedence?

> How do we create the trigger column in drake_plan()?

Unfortunately, `drake_plan()` just deals with targets and commands. Everything else is manual and post-hoc. It is definitely a point in favor of `always()`.

But I think controlling triggers with `always()` is a separate issue from this one. It opens up many more decisions about the interface, whereas `ignore()` is really simple.

**Comment by wlandau** [comment]
Fixed via #317.
